cou/apps/base.py

```diff
@@ -395,6 +387,18 @@ def post_upgrade_steps(
             self._get_reached_expected_target_step(target, units),
         ]
 
+    def upgrade_plan_sanity_checks(self, target: OpenStackRelease) -> None:
+        """Run sanity checks before generating upgrade plan.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :raises ApplicationError: When enable-auto-restarts is not enabled.
+        :raises HaltUpgradePlanGeneration: When the application halt the upgrade plan generation.
+        """
+        self._check_application_target(target)
+        self._check_auto_restarts()
+        logger.info("%s application met all the necessary prerequisites", self.name)
```

Comment:
Non-blocker: I think this log it's hard to understand without a little more context.
```suggestion
        logger.info("%s application met all the necessary prerequisites to generate the upgrade plan", self.name)
```